### ISSUE TITLE
MD024: `allow_different_nesting` should be more permissive

### ISSUE THREAD

**Issue by lukeify** [markdownlint/markdownlint#505]
## Describe the Enhancement:

Currently, the `allow_different_nesting` option for `MD024` only allows duplicate header names if those headers are on the same level. This is the example given in the docs:

```markdown
# Change log

## 2.0.0

### Bug fixes

### Features

## 1.0.0

### Bug fixes
```

In this case, `Bug fixes` is an allowed `h3` header because it has a different `h2` parent header within the heading tree structure. However, with this option enabled, having a duplicate header name as a parent-sibling within the header tree is still disallowed:

```markdown
# Top level header

## Secondary

### Duplicate header 

## Another secondary

### Tertiary

#### Duplicate header
```

In this scenario, the `h3` & `h4` headers named `Duplicate header` are still disallowed even with `allow_different_nesting` set to `true`. Frankly I consider this a bug, as the intent behind `allow_different_nesting` is more a consideration that anchor names could be concatenated from their parents for differentiation, and this would hold true for duplicate header names at differing levels within the header tree structure.  

### Impacted Rules:

* MD024 only.

## Describe the Need:

To allow for duplicate header names at differing levels within the markdown header structure.

## Current Alternative

There is no current alternative. Implementing this would be technically a breaking change, but this project does not appear to adhere to semver so could likely be published as `0.14.0`.

## Can We Help You Implement This?:

I am happy to implement this by refactoring the handling of `allow_different_nesting`. However my main concern is that this project appears only very lightly maintained, and I would appreciate some level of indication that if a PR was submitted, it would be merged in a timely fashion and a release published shortly thereafter.
